airflow/operators/trigger_dagrun.py

```diff
@@ -185,3 +215,39 @@ def execute(self, context: Context):
                 if state in self.allowed_states:
                     self.log.info("%s finished with allowed state %s", self.trigger_dag_id, state)
                     return
+
+    @provide_session
+    def execute_complete(self, context: Context, session: "Session", **kwargs):
```

Comment:
```suggestion
    def execute_complete(self, context: Context, session: Session = NEW_SESSION, **kwargs):
```

Reply:
I don't understand why you're suggesting this over the `@provide_session` decorator